doc/man3/SSL_add1_expected_peer_rpk.pod

```diff
@@ -0,0 +1,93 @@
+=pod
+
+=head1 NAME
+
+SSL_get0_peer_rpk,
+SSL_rpk_send_negotiated,
+SSL_rpk_receive_negotiated,
+SSL_SESSION_get0_peer_rpk,
+SSL_add1_expected_peer_rpk - raw public key (RFC7250) support
+
+=head1 SYNOPSIS
+
+ #include <openssl/ssl.h>
+
+ EVP_PKEY *SSL_get0_peer_rpk(const SSL *s);
+ EVP_PKEY *SSL_SESSION_get0_peer_rpk(const SSL_SESSION *ss);
+ int SSL_rpk_send_negotiated(const SSL *s);
+ int SSL_rpk_receive_negotiated(const SSL *s);
+ int SSL_add1_expected_peer_rpk(SSL *s, EVP_PKEY *pkey);
+
+=head1 DESCRIPTION
+
+SSL_get0_peer_rpk() returns the peer's raw public key from SSL B<s>.
+
+SSL_rpk_send_negotiated() indicates whether sending a raw public key was
+negotiated with the peer.
+
+SSL_rpk_receive_negotiated() indicates whether receiving a raw public key
+was negotiated with the peer.
+
+SSL_SESSION_get0_peer_rpk() returns the peer's raw public key from
+SSL_SESSION B<ss> in the case of resumption.
+
+SSL_add1_expected_peer_rpk() configures the given B<pkey>
+as a public key that the peer is expected to present after negotiating
+raw public keys (RFC7250). The reference count on the provided B<pkey> is
+incremented when added to the SSL B<s>.
+
+=head1 NOTES
+
+Raw public keys are used in place of certificates when the option is
+negotiated. Adding more than one key is possible to allow for key rotation,
+where a peer might be expected to offer an "old" or "new" key and the endpoint
+must be able to accept either one.  It also enables the use of different
+types of keys along with signature-algorithm negotiation.
+
+When raw public keys are used, the certificate verify callback is not called.
+Raw public keys have no subject, issuer, validity dates nor digital signature.
+The peer's raw public key must be added via SSL_add1_expected_peer_rpk()
+for the handshake to succeed. Multiple public keys may be added to the SSL B<s>.
```

Comment:
However, callers may still want to define a new callback for logging or other interaction with the presented key during the handshake, and in particular to override the "ok=0" value to "ok = 1" to continue despite the key not matching any of those provided.  The session verification status should be set to 1 if a key matched and 0 otherwise, and this will then be still be known on resumption.

So a minimal RPK callback would still be useful, though it could perhaps be just a boolean to enable unauthenticated connections, with any logging of the key seen, done post-handshake, but that precludes logging of failed keys when authentication is required.